resolwe/flow/filters.py

```diff
@@ -453,3 +464,81 @@ def get_ordering(self, request, queryset, view):
             return None
 
         return self.get_default_ordering(view)
+
+
+class AnnotationFieldFilter(BaseResolweFilter):
+    """Filter the AnnotationField endpoint."""
+
+    collection_id = filters.CharFilter(method="filter_by_collection")
+
+    def filter_by_collection(self, queryset, name, collection_id):
+        """Filter by the collection id.
+
+        The filtering must also take permissions on the collections in account.
+        """
+        if collection_id is not None:
+            collection = (
+                Collection.objects.filter(pk=collection_id)
+                .filter_for_user(self.request.user)
+                .first()
+            )
+            queryset = queryset.filter(collections=collection)
```

Comment:
This will return all unused annotation fields if the user has no permission to view the collection or the collection doesn't exist. 

Add a check whether the `collection` is `None` and return `.none()` in that case.

Reply:
Done.